File: src/main/java/gay/skitbet/jackiro/command/impl/SetupCommand.java
package gay.skitbet.jackiro.command.impl;

import gay.skitbet.jackiro.command.Command;
import gay.skitbet.jackiro.command.CommandContext;
import gay.skitbet.jackiro.managers.SetupManager;
import gay.skitbet.jackiro.utils.JackiroEmbed;
import gay.skitbet.jackiro.utils.SetupSession;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.interactions.commands.build.OptionData;

import java.util.List;

public class SetupCommand extends Command {
    public SetupCommand() {
        super("setup", "Run the quick setup on Jackiro for this Guild!", Permission.ADMINISTRATOR, null);
    }

    @Override
    public void execute(CommandContext context) {
        SetupSession session = SetupManager.getSetup(context.getGuild().getId());
        if (session != null) {
            context.reply(new JackiroEmbed().error("A setup is already in progress!"));
            return;
        }

        SetupManager.startSetup(context);

    }

    @Override
    public List<OptionData> addOptions(List<OptionData> options) {
        return List.of();
    }
}

File: src/main/kotlin/gay/skitbet/jackiro/command/impl/SetupCommand.kt
package gay.skitbet.jackiro.command.impl

import gay.skitbet.jackiro.command.Command
import gay.skitbet.jackiro.command.CommandContext
import gay.skitbet.jackiro.managers.SetupManager.getSetup
import gay.skitbet.jackiro.managers.SetupManager.startSetup
import gay.skitbet.jackiro.utils.JackiroEmbed
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import java.util.List

class SetupCommand : Command("setup", "Run the quick setup on Jackiro for this Guild!", Permission.ADMINISTRATOR, null, true) {
    override fun execute(context: CommandContext) {
        val session = getSetup(context.guild!!.getId())
        if (session != null) {
            context.reply(JackiroEmbed().error("A setup is already in progress!"))
            return
        }

        startSetup(context)
    }

    override fun addOptions(options: MutableList<OptionData?>): MutableList<OptionData?> {
        return mutableListOf()
    }
}
